modules/Users/metadata/detailviewdefs.php

```diff
@@ -49,7 +49,9 @@
                      * To customise in an upgrade safe way, You need to create custom view instead.
                      * Then override UsersViewDetail::preDisplay().
                      */
-                    'buttons' => array (),
+                    'buttons' => array (
+                        array('customCode' => '<input title="'.translate('LBL_GENERATE_PASSWORD_BUTTON_TITLE','Users').'" class="button" LANGUAGE=javascript onclick="generatepwd(\'{$fields.id.value}\');" type="button" name="password" value="'.translate('LBL_GENERATE_PASSWORD_BUTTON_LABEL','Users').'">"')
```

Comment:
In it's previous point it was caching by following the above comments - following the above comments results in the system caching the id into the template, which is why it was moved here so it was a constant check and not saved into the tpl... So we'd be back to square one.

Suggestions for a resolve without the above solution or the commented recommendation?